### PR TITLE
Refactoring/learning activity

### LINKED ISSUE #95: Make learning activity more robust and make its tests more robust

### PR DESCRIPTION
**Pull request by Nitwix**
This PR refactors LearningActivity and LearningPresenter to be more robust to allow for changes in the future.

Draft because:

- [x] Still need to do some refactorings
- [x] Inject dummy in tests

### CODE REVIEW

File: app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningTest.kt [hunk 1]
```diff
@@ -29,10 +28,10 @@ import org.junit.runner.RunWith
 @HiltAndroidTest
```
Comment (bbjubjub2494) [inline app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningTest.kt:33]:
```suggestion
@HiltAndroidTest
@UninstallModules(DatabaseManagementModule::class)
```
(1/2)

File: app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningTest.kt [hunk 2]
```diff
@@ -29,10 +28,10 @@ import org.junit.runner.RunWith
...
 @RunWith(AndroidJUnit4::class)
 class LearningTest {
-    @get:Rule(order=0)
+    @get:Rule(order = 0)
     val hiltRule = HiltAndroidRule(this)
```
Comment (bbjubjub2494) [inline app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningTest.kt:38]:
```suggestion

    @BindValue
    @Demo2Database
    val dbMgt: DatabaseManagement = DummyDatabaseManagement(DummyDatabaseService())
```
(2/2)

**Review by bbjubjub2494 — CHANGES_REQUESTED** [review]
As requested on Discord.

Also please take the opportunity to remove the `FlakyTest` annotation.

Reference: https://developer.android.com/training/dependency-injection/hilt-testing#binding-new

File: app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningTest.kt (hunk 1 above)
Comment (Nitwix) [reply app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningTest.kt:33]: Thanks a lot!!! It's precisely what I was looking for! And I just tested, it works :D

File: app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningTest.kt [hunk 3]
```diff
@@ -25,14 +30,18 @@ import org.junit.Test
 import org.junit.runner.RunWith
 
 // TODO: fix
-@FlakyTest
+//@FlakyTest
```
Comment (bbjubjub2494) [inline app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningTest.kt]: Don't comment out. Remove the annotation and the TODO altogether

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningActivity.kt [hunk 4]
```diff
@@ -125,19 +146,25 @@ class LearningActivity : AppCompatActivity() {
...
+     */
     private fun dropCallback(event: DragEvent, v: View): Boolean {
         val item: ClipData.Item = event.clipData.getItemAt(0)
-        v.alpha = opaque
-        v.invalidate()
-        Log.d("dropCallback", "${item.text} vs ${v.contentDescription}")
+        setOpacity(v, opaque)
+        Log.d(localClassName, "dropped : ${item.text} on category: ${v.contentDescription}")
+
+        // the classification is considered correct if the text carried by the drag
```
Comment (bbjubjub2494) [inline app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningActivity.kt:162]: That was always kind of a fragile hack right? Does it work with `FirestoreCategorizedPicture` now?

**Review by bbjubjub2494 — CHANGES_REQUESTED** [review]
Just one nitpick and a question but I anticipate I should be able to give the green light very quickly. Good Job!

File: app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningTest.kt (hunk 3 above)
Comment (Nitwix) [reply app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningTest.kt]: yep, will do

**Review by NicolasVial — APPROVED** [review]
Nothing to say for me, good job.

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningActivity.kt (hunk 4 above)
Comment (Nitwix) [reply app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/view/learning/LearningActivity.kt:162]: I realized now that FirestoreCategorizedPicture breaks my usage of contentDescription.
Will be changed in next sprint

**Review by bbjubjub2494 — APPROVED** [review]

**Pull request merged**